### ISSUE TITLE
Client-provided timestamps that are too far in the past are rewritten with the current timestamp; confusion ensues

### ISSUE THREAD

**Issue by tkaemming** [getsentry/sentry#6730]
Timestamps that are [too far in the past (30 days)](https://github.com/getsentry/sentry/blob/d8a53c042a0e80e3e3d85796d8173f04d25b3cf6/src/sentry/event_manager.py#L241-L243) are discarded and rewritten as the current timestamp. This can lead to confusing effects when some clients that may legitimately send very old events (mobile SDKs with offline buffering, for example) have their timestamps rewritten.

This can lead to the incorrect conclusion that an issue that previously existed (especially ones that are not isolated to a specific version, such as a client/server incompatibility that was fixed on the server side, DNS routing issues, etc.) and was assumed to be or explicitly marked as resolved has mysteriously started to occur again in the present.

**Comment by tkaemming** [comment]
One approach here that might make sense (this came up in discussion with @mattrobenolt and @alex-hofsteede) is to stop rewriting old timestamps, and setting a hard cap at some point after which the event is rejected as invalid, and also allow the user to optionally define their own cap as an inbound filter if they'd like to be more restrictive

**Comment by vitaliy-ostapchuk93** [comment]
Any chance this can be disabled in the settings? We currently running a trial to switch to sentry and have to move the dumps we already got, which can be really confusing with the updated timestamps.

**Comment by BYK** [comment]
I think @tkaemming's cap idea is pretty good compared to rewriting as the current timestamp. How do we move forward with that solution? Who else should be involved in making this decision?

**Comment by liewegas** [comment]
We are also encountering this problem.  Any chance of getting a fix or workaround?

**Comment by fpacifici** [comment]
@untitaker Is Relay resetting the timestamp to the current timestamp during the normalization phase? Snuba is not doing it.

**Comment by untitaker** [comment]
Yes, Relay is doing it as did the old Python store. What is the behavior we want? Right now there should be a clear error message at the top indicating that the timestamp is rewritten. It does not happen *silently*.

**Comment by Hassan222-ux** [comment]
https://github.com/getsentry/sentry/issues/6730#issuecomment-680772174
